tests/rptest/tests/memory_sampling_test.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.services.admin import Admin
+from rptest.services.cluster import cluster
+from rptest.tests.redpanda_test import RedpandaTest
+
+BOOTSTRAP_CONFIG = {
+    'memory_enable_memory_sampling': True,
+}
+
+
+class MemorySamplingTestTest(RedpandaTest):
+    def __init__(self, *args, **kwargs):
+        rp_conf = BOOTSTRAP_CONFIG.copy()
+
+        super(MemorySamplingTestTest, self).__init__(*args,
+                                                     extra_rp_conf=rp_conf,
+                                                     **kwargs)
+
+        self.admin = Admin(self.redpanda)
+
+    @cluster(num_nodes=1)
+    def test_get_all_stacks(self):
+        """
+        Verify that the sampled_memory_profile GET endpoint serves valid json
+        with some profile data.
+
+        """
+        admin = Admin(self.redpanda)
+        profile = admin.get_sampled_memory_profile()
+
+        assert len(profile) == 2
+        assert 'shard' in profile[0]
+        assert 'profile' in profile[0]
+        assert len(profile[0]['profile']) > 0
+
+    @cluster(num_nodes=1)
+    def test_get_per_shard_stacks(self):
```

Comment:
it looked like in one of the earlier commit messages that the pr also introduced some new diagnostic output into the redpanda log? if so, we should probably test for that functionality. there is a way in ducktape to grep through the logs of a test's nodes (i think there is a helper dangling off of redpanda.py::RedpandaService).

Reply:
Yes sounds good. 

Haven't done that yet as I am still not sure whether we actually want the periodic printing given it gets quite noisy.

Reply:
As per the other comment, we have decided to only log when memory runs tight. I realize I haven't made it easier to write a test for this now. 

Not entirely sure what the best way is.

Reply:
I wasn't planning to make them configurable as I don't think there is much point in tweaking these. 

Is adding config properties just for tests something we do? I also see a couple instances where we read overrides from an env var? Is that still an accepted thing or frowned upon?

Reply:
>overrides from an env var? Is that still an accepted thing or frowned upon?

it's ok to this, but a little ugly. it's not very common because often folks just add a config option, but it doesn't work in all cases. i wonder if you could boot redpanda with very little memory, and then hit with a big workload?

Reply:
> i wonder if you could boot redpanda with very little memory, and then hit with a big workload?

I will need to give it a try but it sounds very flaky. Any future increase/decrease in "base" memory usage could potentially brake the test.

Will need to sleep on it.

Reply:
@StephanDollberg btw, i forgot to mention, testing for the log message isn't a blocker for this PR. it sounds like it isn't a big part of the experience of using the feature, and as long as we manually test it, it should fine as-is.

Reply:
Added a unit test now. Capturing the log there, added overrides to the service constructor and then simply allocate and wait.

I think this should be stable and not be too hacky.

Let me know what you think.

Reply:
The unit test is cool. We often call back to ducktape because it mostly just works and the hooks are already there to read the logs, hit endpoints, etc, but unit tests are nice when we can go that route.